feat(oxfmt): Respect `.gitignore` in sub directries

leaysgur opened the pull request:
Fixes #17341

Since the `ignore` crate already has this capability, enable it.

Other parts are also adjustable, but for now, only address the specific requests made in the issue.

Boshen commented:
### Merge activity

* **Dec 25, 4:37 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 25, 4:37 AM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

leaysgur committed:
feat(oxfmt): Respect `.gitignore` in sub directries (#17352)

Fixes #17341

Since the `ignore` crate already has this capability, enable it.

Other parts are also adjustable, but for now, only address the specific requests made in the issue.
